backend/src/app/services/parcelDescriptions/parcelDescriptions.service.ts

```diff
@@ -236,7 +245,108 @@ export class ParcelDescriptionsService {
+    const subdivisions: Subdivisions[] = parcelDescriptions.map(
+      (parcelDescription) => {
+        const pid =
+          parcelDescription.descriptionType === ParcelDescriptionType.ParcelID
+            ? parcelDescription.idPinNumber
+            : null;
+        const pin =
+          parcelDescription.descriptionType ===
+          ParcelDescriptionType.CrownLandPIN
+            ? parcelDescription.idPinNumber
+            : null;
+        const crownLandsFileNo =
+          parcelDescription.descriptionType ===
+          ParcelDescriptionType.CrownLandFileNumber
+            ? parcelDescription.idPinNumber
+            : null;
+        return {
+          pid: pid,
+          pin: pin,
+          crownLandsFileNo: crownLandsFileNo,
+          dateNoted: parcelDescription.dateNoted,
+          srAction: parcelDescription.srAction,
+          userAction: parcelDescription.userAction,
+          whoUpdated: userInfo?.givenName,
+          whenUpdated: now,
+          whoCreated: userInfo?.givenName,
+          whenCreated: now,
+          pidStatusCd: 'N', // TODO: Determine what this is and set appropriately.
+        } as Subdivisions;
+      },
+    );
+
+    // Insert the new subdivisions into the database.
+    let insertResult: InsertResult;
+    try {
+      // There is a bug with TypeORM where it will only return columns for the
+      // first parent entity of an entity that uses inheritence. Here I'm using
+      // the querybuilder to work around it.
+      insertResult = await this.subdivisionsRepository
+        .createQueryBuilder()
+        .insert()
+        .into(Subdivisions)
+        .values(subdivisions)
+        .returning('*')
+        .execute();
+    } catch (error) {
+      this.sitesLogger.error(
+        'Exception occured in parcelDescriptionService.addParcelDescriptionsForSite() end',
+        JSON.stringify(error),
+      );
+      throw new BadRequestException('Failed to add Parcel Description.');
+    }
+    const newSubdivisions = insertResult.generatedMaps as Subdivisions[];
+
+    // Use the id from the newly created subdivisions to create the related
+    // siteSubdivisions.
+    const siteSubdivisions: SiteSubdivisions[] = newSubdivisions.map(
+      (newSubdivision) => {
+        return {
+          siteId: siteId,
+          subdivId: newSubdivision.id,
+          dateNoted: newSubdivision.dateNoted,
+          userAction: newSubdivision.userAction,
+          srAction: newSubdivision.userAction,
+          whoUpdated: newSubdivision.whoUpdated,
+          whenUpdated: newSubdivision.whenUpdated,
+          whoCreated: newSubdivision.whoCreated,
+          whenCreated: newSubdivision.whenCreated,
+          initialIndicator: 'N', // TODO: Determine what this is an set appropriately.
+          sendToSr: newSubdivision.srAction === 'pending' ? 'Y' : 'N', // TODO: Validate that this is the correct mapping (I don't think we'll actually use this column).
```

Comment:
if you are using sendToSr property to show the checkbox state on frontend, i would recommend to use true and false instead on Y and N

Reply:
It's not used at all, I'm setting the value based on the existing data, but we don't know what it's used for and the existing database schema has it set to NOT NULL.